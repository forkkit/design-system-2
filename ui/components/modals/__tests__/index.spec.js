// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license
/* eslint-env jest */

import React from 'react';
import createHelpers from '../../../../jest.helpers';
import {
  Taglines,
  ModalSizes,
  Directional,
  Headless,
  Footless,
  HiddenFooter,
  Menu
} from '../base/example';

const { matchesMarkup } = createHelpers(__dirname);

describe('Modal', () => {
  it('renders a modal with a tagline', () => matchesMarkup(<Taglines />));

  it('renders a model that is small', () =>
    matchesMarkup(<ModalSizes size="small" />));

  it('renders a model that is medium', () =>
    matchesMarkup(<ModalSizes size="medium" />));

  it('renders a model that is large', () =>
    matchesMarkup(<ModalSizes size="large" />));

  it('renders a model that is directional', () =>
    matchesMarkup(<Directional />));

  it('renders a model that is headless', () => matchesMarkup(<Headless />));

  it('renders a model that is Footless', () => matchesMarkup(<Footless />));

  it('renders a model that is has a hidden footer', () =>
    matchesMarkup(<HiddenFooter />));

  it('renders a model that is contains a menu', () => matchesMarkup(<Menu />));
});
