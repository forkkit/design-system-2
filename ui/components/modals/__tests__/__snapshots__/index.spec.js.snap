// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders a modal with a tagline 1`] = `
Array [
  <section
    aria-describedby="modal-content-id-1"
    aria-labelledby="modal-heading-01"
    aria-modal="true"
    className="slds-modal slds-fade-in-open"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="slds-modal__container"
    >
      <header
        className="slds-modal__header"
      >
        <button
          className="slds-button slds-button_icon slds-modal__close slds-button_icon-inverse"
          onClick={[Function]}
          title="Close"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon slds-button__icon_large"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Close
          </span>
        </button>
        <h2
          className="slds-modal__title slds-hyphenate"
          id="modal-heading-01"
        >
          Modal header
        </h2>
        <p
          className="slds-m-top_x-small"
        >
          Here’s a tagline if you need it. It is allowed to extend across mulitple lines, so I’m making up content to show that to you. It is allowed to
           
          <a
            href="javascript:void(0);"
          >
            contain links or be a link
          </a>
          .
        </p>
      </header>
      <div
        className="slds-modal__content slds-p-around_medium"
        id="modal-content-id-1"
      >
        <p>
          Sit nulla est ex deserunt exercitation anim occaecat. Nostrud ullamco deserunt aute id consequat veniam incididunt duis in sint irure nisi. Mollit officia cillum Lorem ullamco minim nostrud elit officia tempor esse quis. Cillum sunt ad dolore quis aute consequat ipsum magna exercitation reprehenderit magna. Tempor cupidatat consequat elit dolor adipisicing.
        </p>
        <p>
          Dolor eiusmod sunt ex incididunt cillum quis nostrud velit duis sit officia. Lorem aliqua enim laboris do dolor eiusmod officia. Mollit incididunt nisi consectetur esse laborum eiusmod pariatur proident. Eiusmod et adipisicing culpa deserunt nostrud ad veniam nulla aute est. Labore esse esse cupidatat amet velit id elit consequat minim ullamco mollit enim excepteur ea.
        </p>
      </div>
      <footer
        className="slds-modal__footer"
      >
        <button
          className="slds-button slds-button_neutral"
        >
          Cancel
        </button>
        <button
          className="slds-button slds-button_brand"
        >
          Save
        </button>
      </footer>
    </div>
  </section>,
  <div
    className="slds-backdrop slds-backdrop_open"
  />,
]
`;

exports[`Modal renders a model that is Footless 1`] = `
Array [
  <section
    aria-describedby="modal-content-id-1"
    aria-labelledby="modal-heading-01"
    aria-modal="true"
    className="slds-modal slds-fade-in-open"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="slds-modal__container"
    >
      <header
        className="slds-modal__header"
      >
        <button
          className="slds-button slds-button_icon slds-modal__close slds-button_icon-inverse"
          onClick={[Function]}
          title="Close"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon slds-button__icon_large"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Close
          </span>
        </button>
        <h2
          className="slds-modal__title slds-hyphenate"
          id="modal-heading-01"
        >
          Modal header
        </h2>
      </header>
      <div
        className="slds-modal__content slds-p-around_medium"
        id="modal-content-id-1"
      >
        <p>
          Sit nulla est ex deserunt exercitation anim occaecat. Nostrud ullamco deserunt aute id consequat veniam incididunt duis in sint irure nisi. Mollit officia cillum Lorem ullamco minim nostrud elit officia tempor esse quis. Cillum sunt ad dolore quis aute consequat ipsum magna exercitation reprehenderit magna. Tempor cupidatat consequat elit dolor adipisicing.
        </p>
        <p>
          Dolor eiusmod sunt ex incididunt cillum quis nostrud velit duis sit officia. Lorem aliqua enim laboris do dolor eiusmod officia. Mollit incididunt nisi consectetur esse laborum eiusmod pariatur proident. Eiusmod et adipisicing culpa deserunt nostrud ad veniam nulla aute est. Labore esse esse cupidatat amet velit id elit consequat minim ullamco mollit enim excepteur ea.
        </p>
      </div>
    </div>
  </section>,
  <div
    className="slds-backdrop slds-backdrop_open"
  />,
]
`;

exports[`Modal renders a model that is contains a menu 1`] = `
Array [
  <section
    aria-describedby="modal-content-unique-id-2"
    aria-labelledby="modal-heading-unique-id-1"
    aria-modal="true"
    className="slds-modal slds-fade-in-open"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="slds-modal__container"
    >
      <header
        className="slds-modal__header"
      >
        <button
          className="slds-button slds-button_icon slds-modal__close slds-button_icon-inverse"
          onClick={[Function]}
          title="Close"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon slds-button__icon_large"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Close
          </span>
        </button>
        <h2
          className="slds-modal__title slds-hyphenate"
          id="modal-heading-unique-id-1"
        >
          Modal header
        </h2>
      </header>
      <div
        className="slds-modal__menu"
        id="modal-content-unique-id-2"
      >
        <p>
          Sit nulla est ex deserunt exercitation anim occaecat. Nostrud ullamco deserunt aute id consequat veniam incididunt duis in sint irure nisi. Mollit officia cillum Lorem ullamco minim nostrud elit officia tempor esse quis. Cillum sunt ad dolore quis aute consequat ipsum magna exercitation reprehenderit magna. Tempor cupidatat consequat elit dolor adipisicing.
        </p>
        <p>
          Dolor eiusmod sunt ex incididunt cillum quis nostrud velit duis sit officia. Lorem aliqua enim laboris do dolor eiusmod officia. Mollit incididunt nisi consectetur esse laborum eiusmod pariatur proident. Eiusmod et adipisicing culpa deserunt nostrud ad veniam nulla aute est. Labore esse esse cupidatat amet velit id elit consequat minim ullamco mollit enim excepteur ea.
        </p>
      </div>
      <footer
        className="slds-modal__footer slds-hide"
      />
    </div>
  </section>,
  <div
    className="slds-backdrop slds-backdrop_open"
  />,
]
`;

exports[`Modal renders a model that is directional 1`] = `
Array [
  <section
    aria-describedby="modal-content-id-1"
    aria-labelledby="modal-heading-01"
    aria-modal="true"
    className="slds-modal slds-fade-in-open"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="slds-modal__container"
    >
      <header
        className="slds-modal__header"
      >
        <button
          className="slds-button slds-button_icon slds-modal__close slds-button_icon-inverse"
          onClick={[Function]}
          title="Close"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon slds-button__icon_large"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Close
          </span>
        </button>
        <h2
          className="slds-modal__title slds-hyphenate"
          id="modal-heading-01"
        >
          Modal header
        </h2>
      </header>
      <div
        className="slds-modal__content slds-p-around_medium"
        id="modal-content-id-1"
      >
        <p>
          Sit nulla est ex deserunt exercitation anim occaecat. Nostrud ullamco deserunt aute id consequat veniam incididunt duis in sint irure nisi. Mollit officia cillum Lorem ullamco minim nostrud elit officia tempor esse quis. Cillum sunt ad dolore quis aute consequat ipsum magna exercitation reprehenderit magna. Tempor cupidatat consequat elit dolor adipisicing.
        </p>
        <p>
          Dolor eiusmod sunt ex incididunt cillum quis nostrud velit duis sit officia. Lorem aliqua enim laboris do dolor eiusmod officia. Mollit incididunt nisi consectetur esse laborum eiusmod pariatur proident. Eiusmod et adipisicing culpa deserunt nostrud ad veniam nulla aute est. Labore esse esse cupidatat amet velit id elit consequat minim ullamco mollit enim excepteur ea.
        </p>
      </div>
      <footer
        className="slds-modal__footer slds-modal__footer_directional"
      >
        <button
          className="slds-button slds-button_neutral"
        >
          Skip This Step
        </button>
        <button
          className="slds-button slds-button_brand"
        >
          Save & Next
        </button>
      </footer>
    </div>
  </section>,
  <div
    className="slds-backdrop slds-backdrop_open"
  />,
]
`;

exports[`Modal renders a model that is has a hidden footer 1`] = `
Array [
  <section
    aria-describedby="modal-content-id-1"
    aria-labelledby="modal-heading-01"
    aria-modal="true"
    className="slds-modal slds-fade-in-open"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="slds-modal__container"
    >
      <header
        className="slds-modal__header"
      >
        <button
          className="slds-button slds-button_icon slds-modal__close slds-button_icon-inverse"
          onClick={[Function]}
          title="Close"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon slds-button__icon_large"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Close
          </span>
        </button>
        <h2
          className="slds-modal__title slds-hyphenate"
          id="modal-heading-01"
        >
          Modal header
        </h2>
      </header>
      <div
        className="slds-modal__content slds-modal__content_has-hidden-footer slds-p-around_medium"
        id="modal-content-id-1"
      >
        <p>
          Sit nulla est ex deserunt exercitation anim occaecat. Nostrud ullamco deserunt aute id consequat veniam incididunt duis in sint irure nisi. Mollit officia cillum Lorem ullamco minim nostrud elit officia tempor esse quis. Cillum sunt ad dolore quis aute consequat ipsum magna exercitation reprehenderit magna. Tempor cupidatat consequat elit dolor adipisicing.
        </p>
        <p>
          Dolor eiusmod sunt ex incididunt cillum quis nostrud velit duis sit officia. Lorem aliqua enim laboris do dolor eiusmod officia. Mollit incididunt nisi consectetur esse laborum eiusmod pariatur proident. Eiusmod et adipisicing culpa deserunt nostrud ad veniam nulla aute est. Labore esse esse cupidatat amet velit id elit consequat minim ullamco mollit enim excepteur ea.
        </p>
      </div>
      <footer
        className="slds-modal__footer slds-hide"
      />
    </div>
  </section>,
  <div
    className="slds-backdrop slds-backdrop_open"
  />,
]
`;

exports[`Modal renders a model that is headless 1`] = `
Array [
  <section
    aria-describedby="modal-content-id-1"
    aria-label="Meaningful description of the modal content"
    aria-modal="true"
    className="slds-modal slds-fade-in-open"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="slds-modal__container"
    >
      <header
        className="slds-modal__header slds-modal__header_empty"
      >
        <button
          className="slds-button slds-button_icon slds-modal__close slds-button_icon-inverse"
          onClick={[Function]}
          title="Close"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon slds-button__icon_large"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Close
          </span>
        </button>
      </header>
      <div
        className="slds-modal__content slds-p-around_medium"
        id="modal-content-id-1"
      >
        <p>
          Sit nulla est ex deserunt exercitation anim occaecat. Nostrud ullamco deserunt aute id consequat veniam incididunt duis in sint irure nisi. Mollit officia cillum Lorem ullamco minim nostrud elit officia tempor esse quis. Cillum sunt ad dolore quis aute consequat ipsum magna exercitation reprehenderit magna. Tempor cupidatat consequat elit dolor adipisicing.
        </p>
        <p>
          Dolor eiusmod sunt ex incididunt cillum quis nostrud velit duis sit officia. Lorem aliqua enim laboris do dolor eiusmod officia. Mollit incididunt nisi consectetur esse laborum eiusmod pariatur proident. Eiusmod et adipisicing culpa deserunt nostrud ad veniam nulla aute est. Labore esse esse cupidatat amet velit id elit consequat minim ullamco mollit enim excepteur ea.
        </p>
      </div>
      <footer
        className="slds-modal__footer"
      >
        <button
          className="slds-button slds-button_neutral"
        >
          Cancel
        </button>
        <button
          className="slds-button slds-button_brand"
        >
          Save
        </button>
      </footer>
    </div>
  </section>,
  <div
    className="slds-backdrop slds-backdrop_open"
  />,
]
`;

exports[`Modal renders a model that is large 1`] = `
Array [
  <section
    aria-describedby="modal-content-id-1"
    aria-labelledby="modal-heading-01"
    aria-modal="true"
    className="slds-modal slds-fade-in-open slds-modal_large"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="slds-modal__container"
    >
      <header
        className="slds-modal__header"
      >
        <button
          className="slds-button slds-button_icon slds-modal__close slds-button_icon-inverse"
          onClick={[Function]}
          title="Close"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon slds-button__icon_large"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Close
          </span>
        </button>
        <h2
          className="slds-modal__title slds-hyphenate"
          id="modal-heading-01"
        >
          Modal header
        </h2>
      </header>
      <div
        className="slds-modal__content slds-p-around_medium"
        id="modal-content-id-1"
      >
        <p>
          Sit nulla est ex deserunt exercitation anim occaecat. Nostrud ullamco deserunt aute id consequat veniam incididunt duis in sint irure nisi. Mollit officia cillum Lorem ullamco minim nostrud elit officia tempor esse quis. Cillum sunt ad dolore quis aute consequat ipsum magna exercitation reprehenderit magna. Tempor cupidatat consequat elit dolor adipisicing.
        </p>
        <p>
          Dolor eiusmod sunt ex incididunt cillum quis nostrud velit duis sit officia. Lorem aliqua enim laboris do dolor eiusmod officia. Mollit incididunt nisi consectetur esse laborum eiusmod pariatur proident. Eiusmod et adipisicing culpa deserunt nostrud ad veniam nulla aute est. Labore esse esse cupidatat amet velit id elit consequat minim ullamco mollit enim excepteur ea.
        </p>
      </div>
      <footer
        className="slds-modal__footer"
      >
        <button
          className="slds-button slds-button_neutral"
        >
          Cancel
        </button>
        <button
          className="slds-button slds-button_brand"
        >
          Save
        </button>
      </footer>
    </div>
  </section>,
  <div
    className="slds-backdrop slds-backdrop_open"
  />,
]
`;

exports[`Modal renders a model that is medium 1`] = `
Array [
  <section
    aria-describedby="modal-content-id-1"
    aria-labelledby="modal-heading-01"
    aria-modal="true"
    className="slds-modal slds-fade-in-open slds-modal_medium"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="slds-modal__container"
    >
      <header
        className="slds-modal__header"
      >
        <button
          className="slds-button slds-button_icon slds-modal__close slds-button_icon-inverse"
          onClick={[Function]}
          title="Close"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon slds-button__icon_large"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Close
          </span>
        </button>
        <h2
          className="slds-modal__title slds-hyphenate"
          id="modal-heading-01"
        >
          Modal header
        </h2>
      </header>
      <div
        className="slds-modal__content slds-p-around_medium"
        id="modal-content-id-1"
      >
        <p>
          Sit nulla est ex deserunt exercitation anim occaecat. Nostrud ullamco deserunt aute id consequat veniam incididunt duis in sint irure nisi. Mollit officia cillum Lorem ullamco minim nostrud elit officia tempor esse quis. Cillum sunt ad dolore quis aute consequat ipsum magna exercitation reprehenderit magna. Tempor cupidatat consequat elit dolor adipisicing.
        </p>
        <p>
          Dolor eiusmod sunt ex incididunt cillum quis nostrud velit duis sit officia. Lorem aliqua enim laboris do dolor eiusmod officia. Mollit incididunt nisi consectetur esse laborum eiusmod pariatur proident. Eiusmod et adipisicing culpa deserunt nostrud ad veniam nulla aute est. Labore esse esse cupidatat amet velit id elit consequat minim ullamco mollit enim excepteur ea.
        </p>
      </div>
      <footer
        className="slds-modal__footer"
      >
        <button
          className="slds-button slds-button_neutral"
        >
          Cancel
        </button>
        <button
          className="slds-button slds-button_brand"
        >
          Save
        </button>
      </footer>
    </div>
  </section>,
  <div
    className="slds-backdrop slds-backdrop_open"
  />,
]
`;

exports[`Modal renders a model that is small 1`] = `
Array [
  <section
    aria-describedby="modal-content-id-1"
    aria-labelledby="modal-heading-01"
    aria-modal="true"
    className="slds-modal slds-fade-in-open slds-modal_small"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="slds-modal__container"
    >
      <header
        className="slds-modal__header"
      >
        <button
          className="slds-button slds-button_icon slds-modal__close slds-button_icon-inverse"
          onClick={[Function]}
          title="Close"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon slds-button__icon_large"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Close
          </span>
        </button>
        <h2
          className="slds-modal__title slds-hyphenate"
          id="modal-heading-01"
        >
          Modal header
        </h2>
      </header>
      <div
        className="slds-modal__content slds-p-around_medium"
        id="modal-content-id-1"
      >
        <p>
          Sit nulla est ex deserunt exercitation anim occaecat. Nostrud ullamco deserunt aute id consequat veniam incididunt duis in sint irure nisi. Mollit officia cillum Lorem ullamco minim nostrud elit officia tempor esse quis. Cillum sunt ad dolore quis aute consequat ipsum magna exercitation reprehenderit magna. Tempor cupidatat consequat elit dolor adipisicing.
        </p>
        <p>
          Dolor eiusmod sunt ex incididunt cillum quis nostrud velit duis sit officia. Lorem aliqua enim laboris do dolor eiusmod officia. Mollit incididunt nisi consectetur esse laborum eiusmod pariatur proident. Eiusmod et adipisicing culpa deserunt nostrud ad veniam nulla aute est. Labore esse esse cupidatat amet velit id elit consequat minim ullamco mollit enim excepteur ea.
        </p>
      </div>
      <footer
        className="slds-modal__footer"
      >
        <button
          className="slds-button slds-button_neutral"
        >
          Cancel
        </button>
        <button
          className="slds-button slds-button_brand"
        >
          Save
        </button>
      </footer>
    </div>
  </section>,
  <div
    className="slds-backdrop slds-backdrop_open"
  />,
]
`;
